### ISSUE TITLE
[General Question] Warning for each list item in a win32 list control in report mode 

### ISSUE THREAD

**Issue by skuskusku** [microsoft/accessibility-insights-windows#1843]
**How do I handle the warning I get for each list control item in a win32 list control in report mode about non-conforming structure**

My application uses a pretty standard win32 list control (syslistview32) from comctl32 in report mode. If I run AI4Win tests against it, I get a warning for each and every list control item about a seemingly non-conforming structure. The warning message text is the following:

"The given element is expected to have the following structure: ListItem and NoChild(IsContentElement)."

The suggested fix goes like this: "Set the element to conform to the following structure: ListItem and NoChild(IsContentElement)."

I do not have the faintest idea what AI4Win is telling me here, can please somebody explain? Is there something I can do to avoid this warning?

Any help appreciated,

-- 
Stefan

**Comment by v-viyada** [comment]
@skuskusku Thanks for reaching out. In order for us to investigate this issue we need some way to replicate it. Can you provide a sample app that shows the behavior as well as the a11ytest file if you captured it? This will allow us to investigate what may be happening in this scenario. Thanks!

**Comment by skuskusku** [comment]
I have uploaded a sample app here: https://mcblogs.craalse.de/skufiles/2024/jul/listctrltest_1843.zip

In order to build you have to have MFC with static libraries installed with VS2022. If you run an AI4Win test on the samples dialog, it will pass without errors. Now look into the test results in the automation pane and drill down from the list control to its first item with the text 

list item 'Row 0 Column 0'

in the automation tree and click on the link labeled "Passed and Uncertain tests (24)" and you will see that this element shows the warning and suggested fix text I mentioned in my initial text for this issue.

The .a11ytest file is here:

https://mcblogs.craalse.de/skufiles/2024/jul/1843.a11ytest

**Comment by codeofdusk** [comment]
> Passed and Uncertain tests

The element structure tests do fall into the category of "uncertain" since they serve as guidelines of good UI Automation (UIA) provider implementations and are sometimes useful, but they are subject to a higher rate of false positives, and it would be unreasonable to expect that applications pass all tests marked "uncertain".

In your particular case of a `Win32` `syslist32` control, the UIA provider implementation is sadly less than perfect (for instance, [NVDA doesn't use it, falling back to MSAA](https://github.com/nvaccess/nvda/pull/15799)). This is unlikely to change in a significant way to maintain backward compatibility.

In short, this error isn't your fault, and you can safely ignore it.
